### PR TITLE
feat: add config to enable prettified changelogs

### LINKED ISSUE #3: Prettified Changelogs including emojis
Hi @christophwitzko,

you might remember that I once asked, if you also support templates for the generated changelogs. I now thought about it and I am planning to create a PR for this. My intended behavior would be, to extend the `changelog-generator-default` plugin with some config to export prettified changelogs if intended. This could be managed by extending the `ChangelogGeneratorConfig` accordingly. 
```
changelogRes := changelogGenerator.Generate(&generator.ChangelogGeneratorConfig{
	Commits:       commits,
	LatestRelease: release,
	NewVersion:    newVer,
        Prettify: true | false, // needs to be provided with the semantic-release execution, default = false
})
```
[ChangelogGeneratorConfig](https://github.com/go-semantic-release/semantic-release/blob/master/cmd/semantic-release/main.go#L221:L225)

This config would then be part of the changelog generation. In the generator itself I would mainly extend the `ChangelogType` to also support a prefixed emoji:
```
type ChangelogType struct {
	Type    string
	Text    string
	Content string
        Emoji string
}
```
[ChangelogType](https://github.com/go-semantic-release/changelog-generator-default/blob/master/pkg/generator/changelog_types.go#L3:L7)

Depending on the ~~`prettify`~~ `emojis` flag it would then prefix the changelog headings with some emoji.

A potential Example would then look like the following:

# 1.0.0 (2020-12-02)

### 🎁 Feature:

    ...

### 🐞 Bug Fixes:

    .....

### 🚧 Chores:

    .....

I will create a PR and in case you also find it a nice feature, feel free to merge when ready ;)

Regards 
Timo

### PR DESCRIPTION
**Pull request by timoknapp**
Hi @christophwitzko ,

As already discussed in #3  here is one of the two PRs regarding the prettified changelogs.

It will simple work with the `changelog-generator-opt` config which can be provided when running semantic-release. If you dont provide the config it wont prettify the changelogs of course.

Here is an example call including the final `CHANGELOG.md`:

```
./semantic-release --provider "gitlab" --token $TOKEN --provider-opt "gitlab_projectid=42" --no-ci --dry --changelog "CHANGELOG.md" --changelog-generator-opt "emojis=true"
[go-semantic-release]: version:
[go-semantic-release]: ci-condition plugin: default@1.2.2
[go-semantic-release]: provider plugin: GitLab@1.4.1
[go-semantic-release]: getting default branch...
[go-semantic-release]: found default branch: master
[go-semantic-release]: found current branch: master
[go-semantic-release]: found current sha: master
[go-semantic-release]: getting latest release...
[go-semantic-release]: found version: 1.1.2
[go-semantic-release]: getting commits...
[go-semantic-release]: analyzing commits...
[go-semantic-release]: commit-analyzer plugin: default@1.3.1
[go-semantic-release]: calculating new version...
[go-semantic-release]: new version: 1.1.3
[go-semantic-release]: generating changelog...
[go-semantic-release]: changelog-generator plugin: default@dev
[go-semantic-release]: changelog-generator prettified_output: true
[go-semantic-release]: DRY RUN: no release was created
```

```
cat CHANGELOG.md

## 1.1.3 (2020-12-08)

#### 🐞 Bug Fixes

* resolve refresh bug (a994ee5c)
```

Feel free to review, I am happy for feedback.

Regards
Timo

### CODE REVIEW

**Comment by christophwitzko** [comment]
Hey @timoknapp, this is great thank you! 😄 

Can we use a more clear name than `prettified_output`? I would prefer something like `emojis` or `emoji_prefix`.

Thanks in advance. ☺️

**Comment by timoknapp** [comment]
Moin @christophwitzko ,

sure I also felt not too happy with the name. Whats your preference? `emojis` or `emoji_prefix`.

Cheers
Timo

**Comment by christophwitzko** [comment]
Let's use `emojis` 👍

**Comment by timoknapp** [comment]
I will update PR's accordingly.

**Comment by christophwitzko** [comment]
Thanks ☺️

File: pkg/generator/changelog_types.go [hunk 1]
```diff
@@ -30,51 +31,63 @@ func (ct *ChangelogTypes) AppendContent(cType, content string) {
 
 var defaultTypes = ChangelogTypes{
 	{
-		Type: "%%bc%%",
-		Text: "Breaking Changes",
+		Type:  "%%bc%%",
+		Text:  "Breaking Changes",
+		Emoji: "📣 ",
```
Comment (christophwitzko) [inline pkg/generator/changelog_types.go]: Can we remove the space character here for all fields and add it when needed with `prettifyPrefix = ct.Emoji + " "`

File: examples/GENERATED_CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,32 @@
...
+
+#### 🐞 Bug Fixes
+
+* commit message (abcd)
+
+#### 📦 Build
+
+* commit message (43218765)
+
+#### 🔁  CI
```
Comment (christophwitzko) [inline examples/GENERATED_CHANGELOG.md]: Here are 2 spaces after the emoji 😇

File: pkg/generator/changelog_types.go (hunk 1 above)
Comment (timoknapp) [reply pkg/generator/changelog_types.go]: will do

File: examples/GENERATED_CHANGELOG.md (hunk 2 above)
Comment (timoknapp) [reply examples/GENERATED_CHANGELOG.md]: spaces are evil

**Comment by timoknapp** [comment]
I updated all the parts, feel free to take a closer look though ;)

File: README.md [hunk 3]
```diff
@@ -21,6 +21,10 @@ The changelog generator will order the types of commits in the changelog in the
 - Build
 - CI
 
+## Prettified changelogs
```
Comment (christophwitzko) [inline README.md]: I would call it `Emoji changelogs`

Comment (timoknapp) [reply README.md]: Done

**Review by christophwitzko — APPROVED** [review]

**Pull request merged**